### PR TITLE
feat: merge formatters and linters

### PR DESCRIPTION
**Pull request by jaredallard**
## What this PR does / why we need it

This PR merges the `linters` and `fmt` scripts, mostly, together. They now share the same `linters/<LANGUAGE>.sh` script for bother linting and formatting.

Misc things:

 * Better non-terminal handling
 * When tool outputs information (in a supported terminal), don't rewrite the line
 * When errored, with no output, append the time then show the error
 * Fixes asdf plugin list issue
 * Fixes `mage help` issue

## Jira ID

[DT-2571] [DT-2572]

## Notes for your reviewers

E2E is failing because I updated the `bootstrap.lock` to get this to pass running on itself, it'll be fixed post-merge

[DT-2571]: https://outreach-io.atlassian.net/browse/DT-2571?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
[DT-2572]: https://outreach-io.atlassian.net/browse/DT-2572?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: root/Makefile [hunk 1]
```diff
@@ -1,16 +1,18 @@
 SHELL                := /usr/bin/env bash
 
+BOOTSTRAP_DIR        := $(shell if [[ "$$(yq -r .name service.yaml)" == "devbase" ]]; then echo "$(CURDIR)" ; else echo "$(CURDIR)/.bootstrap"; fi)
```
Comment (jaredallard) [inline root/Makefile:3]: This enables devbase to run on itself

**Comment by jaredallard** [comment]
OK, I've tested this on `bulkworkflow` and `authz` and it works great. Reasonably confident this is good to merge post-release.

**Review by tonymbugua-outreach — APPROVED** [review]
LGTM

File: shell/linters/bash.sh [hunk 2]
```diff
@@ -7,21 +7,23 @@ SHELLCHECKPATH="$DIR/shellcheck.sh"
 # shellcheck disable=SC2034
 extensions=(sh bash)
 
-shellcheck() {
-  if ! git ls-files '*.sh' | xargs -n40 "$SHELLCHECKPATH" -x -P SCRIPTDIR; then
-    error "shellcheck failed on some files. Run 'make fmt' to fix."
```
Comment (artemchernyak-outreach) [inline shell/linters/bash.sh:12]: I actually really liked having these errors come up. Could we either leave them in or add a top level error like: `Linting faild on some files. Run 'make fmt' to fix.`

**Pull request merged**